File: src/APIs/BibliotecaGamificada.Turmas.Api/Models/TurmaCadastroModel.cs

namespace BibliotecaGamificada.Turmas.Api.Models
{
    public class TurmaCadastroModel
    {
        public string? id { get; set; }
        public string nome { get; set; }
        public int anoLetivo { get; set; }
        public string professor { get; set; }
        public List<string>? alunos { get; set; }
        public List<string>? livros { get; set; }
        public string instituicao { get; set; }

        public TurmaCadastroModel(string nome, int anoLetivo,string professor, List<string> alunos, List<string> livros, string instituicao)
        {
            this.nome = nome;
            this.anoLetivo = anoLetivo;
            this.professor = professor;
            this.alunos = alunos;
            this.livros = livros;
            this.instituicao = instituicao;
        }
    }
}

File: ClientApp/src/app/models/entidades/TurmaCadastro.model.ts

export class TurmaCadastroModel{
    id?: string = "";
    nome: string = "";
    anoLetivo: number= 2022;
    instituicao: string = "";
    professor: string = "";
    livros: Array<string>;
    alunos: Array<string>;
    
    constructor(nome: string, anoLetivo: number, instituicao: string, professor: string) {
        this.nome = nome;
        this.anoLetivo = anoLetivo;
        this.instituicao = instituicao;
        this.professor = professor;
        this.livros = [];
        this.alunos = [];
    }
}